### ISSUE TITLE
JSON ssh test works but Zabbix doesn't.

### ISSUE THREAD

**Issue by Coolbeans007** [patricegautier/unifiZabbix#12]
I've installed it as your documentation says, but when I get to the Zabbix web interface nothing is showing up.
 sudo -u zabbix /usr/lib/zabbix/externalscripts/mca-dump-short.sh -d 192.168.x.x -u xxx-i /root/zb_id_rsa.pub -t UDMP
When executed, this returns the data requested by the script. Unifi is sending the data back correctly. 
I just need help setting up the hosts. I've imported the XML file and all of the templates but I do not know what interface to add to the host. And also do I need to create an ssh item for the main UDMP host that runs the dump script?

**Comment by patricegautier** [comment]
Good news - I think you have done the hard part, and you just have to do the easy part now.

you just have to create one host per unifi device, with the right ip and assign it 2 templates, 'Unifi SSH Host' and the appropriate one for the device type.  For ex, your gateway/udmp should look like:

[router config.pdf](https://github.com/patricegautier/unifiZabbix/files/6065803/router.config.pdf)

[templates.pdf](https://github.com/patricegautier/unifiZabbix/files/6065805/templates.pdf)

The fact the interface is the 'agent interface' is actually not relevant, only the IP matters given the items in the template.

Makes sense?

**Comment by Coolbeans007** [comment]
Yes! Thanks so much for clarifying. I've been working on this for a while and managed to get data from it. I added a host for the UDMP and then linked the appropriate templates to it. The private keys were the issue. You might want to add in your documentation how to add private keys through ssh to the Unifi controller.
I realized that Zabbix wasn't able to connect without the public key being on the UDMP way to late. I had the JSON data very early on from the ssh check your script can run. 
Everything works now which is great. I'm happy to see people integrating Unifi monitoring into Zabbix.
While ssh is a little dirty rather than SNMP it works much better than the nearly impossible to setup SNMP from the beta branch at Ubiquiti.
Thanks for the explanation. I moved from Icinga after spending 4 days wasting my time on impossible tasks. I'm exited to add my whole topology to Zabbix.
